### PR TITLE
sort_by_nil_fix - fix sort_by nil

### PR DESCRIPTION
**Pull request by 7sedam7**

### CODE REVIEW

**Comment by SpComb** [comment]
Hmm, what situation does this happen in? It is actually possible for the `node_number` to be `nil`, but that should only happen for a very short period of time while a node is connecting.

Are you perhaps running into something like the race condition in #2071?

Just a little surprised that this popped up now, right after restoring the original `sort_by` in 1.3.2.

**Comment by 7sedam7** [comment]
Signed CLA.

@SpComb  I think it's not #2071, I do not know how I got to it, but I was not able to get rid of it, so I just changed it for my self and added PR in case it's ok for you as well.

**Comment by SpComb** [comment]
Your git commit author doesn't match up with your GitHub account's email addresses, so the cla-assistant.io check didn't pass yet... you need to either add that email address to your github profile and re-check, or rewrite the commit author and force-push?

File: cli/lib/kontena/cli/nodes/list_command.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def node_data
           end
         end
 
-        grid_nodes.sort_by { |n| n['node_number'] }
+        grid_nodes.sort_by { |n| n['node_number'].to_i }
```
Comment (SpComb) [inline cli/lib/kontena/cli/nodes/list_command.rb]: Could also have a spec case with `{ 'node_number' => nil }` (see `cli/spec/kontena/cli/nodes/list_command_spec.rb`)

**Comment by SpComb** [comment]
@7sedam7 this is still pending on the CLA check passing, because your git commit author does not match up with your github email address.

**Comment by SpComb** [comment]
The server should now be fixed in #2694 to always assign a `node_number` to nodes, so this shouldn't be a problem on the CLI anymore... I hope.

**Pull request closed without merging**